documentation/man/features/kw-signature.rst

```diff
@@ -0,0 +1,94 @@
+===================================================
+kw-signature - Write commit and patch trailer lines
+===================================================
+
+.. _signature-doc:
+
+SYNOPSIS
+========
+| *kw* *signature* (-s(<string>) | \--add-signed-off-by=(<string>)) [\--verbose] [<sha> | <patch>]
+| *kw* *signature* (-r(<string>) | \--add-reviewed-by=(<string>)) [\--verbose] [<sha> | <patch>]
+| *kw* *signature* (-a(<string>) | \--add-acked-by=(<string>)) [\--verbose] [<sha> | <patch>]
+| *kw* *signature* (-f(<sha>) | \--add-fixes=(<sha>)) [\--verbose] [<sha> | <patch>]
+
+DESCRIPTION
+===========
+This is a wrapper to some useful usage of the 'git commit --amend',
+'git rebase' and the 'git interpret-trailers' commands. This kw command
+is able to perform usual operations over trailer lines of commits and patches.
+By default, it uses the commit currently pointed by ``HEAD`` as the target
+of the operation if user does not specify one. Also, no operation is used by
+default. At least one operation option must be given.
+
+This command does not allow repetition of options. For instance, the
+following command will give an error::
+
+  $ kw signature -r"Reviewer Name <mail>" -r"Some Name <mail>"
+
+However, using different options combined is possible. For instance,
+the following command will add **Reviewed-by** and **Signed-off-by**, in
+this order, to the commit's trailers pointed by ``HEAD`` using the
+user's git config::
+
+  $ kw signature -r -s
+
+Changing the order of the options affects the order the lines are written.
+
+OPTIONS
+=======
+-s, \--add-signed-off-by:
+  Adds a **Signed-off-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+  
+-r, \--add-reviewed-by:
+  Adds a **Reviewed-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+
+-a, \--add-acked-by:
+  Adds a **Acked-by** trailer line to either commits or patchsets.
+  By default it uses `user.name` and `user.email` from git config to
+  write these lines if no argument is given to this option.
+
+-f, \--add-fixes:
+  Adds a **Fixes** trailer line to either commits or patchsets.
+  It requires an argument, which must be a valid commit reference, to
+  define the fixed commit's hash that follows this tag.
+
+\--verbose:
+  Display commands executed under the hood.
+
+EXAMPLES
+========
+Adding a **Reviewed-by** line to commit pointed by ``HEAD``::
+
+  $ kw signature --add-reviewed-by='Reviewer Name <reviewer@mail.org>'
+
+Adding a **Acked-by** line to last 3 commits starting from ``HEAD``::
+
+  $ kw signature --acked-by='Some Name <example@mail.com>' HEAD~3
```

Comment:
Also, do the tag options support multiple emails? For example:

```bash
 kw signature --add-reviewed-by='Reviewer Name <reviewer@mail.org>, Another Name <anothername@ljkjsdfal.com>'
```

That results in:

```
Reviewed-by: Reviewer Name <reviewer@mail.org>
Reviewed-by: Another Name <anothername@ljkjsdfal.com>
```

Reply:
Since the command now accepts repetition of options, I'm not sure if supporting multiple signatures in the same argument is required. Also, with the current implementation this will not work beacause the program will accuse `Reviewer Name <reviewer@mail.org>, Another Name <anothername@ljkjsdfal.com>` as an invalid signature format.

However, if we define a **separator**, we can split this argument into multiple signatures and then validate each one.

Supporting multiple emails in the same argument is desirable?

Reply:
yeah, and this behavior looks similar to what we have when using git-send-email.